packages/keyring-controller/src/KeyringController.ts

```diff
@@ -2100,9 +2178,14 @@ export class KeyringController extends BaseController<
       }
 
       const updatedKeyrings = await this.#getUpdatedKeyrings();
+      if (updatedKeyrings.length !== this.#keyringsMetadata.length) {
+        throw new Error(KeyringControllerError.KeyringMetadataLengthMismatch);
+      }
+
       this.update((state) => {
         state.vault = updatedState.vault;
         state.keyrings = updatedKeyrings;
+        state.keyringsMetadata = this.#keyringsMetadata;
```

Comment:
This might be how `#keyringsMetadata` is getting frozen - it's passed directly in as part of state here. Perhaps we can clone it here to avoid that problem

Reply:
Ha 🙌 You are 100% right! Thank you!